### ISSUE TITLE
“Type” & “Choose from list” drop downs are both blank

### ISSUE THREAD

**Issue by ukmoose** [zinen/node-red-contrib-dirigera#2]
Thanks for making this!
Unfortunately when I try to use it I am running into a problem. 

I type in the IP address of the Hub
Click “Get access code”  (access code is received)

“Access code received OK. Ready to save this config”

I click [Add] 
The “Type” & “Choose from list” drop downs both state “Make and deploy config first”

I click [Done] and then [Deploy]
When I then double click on a node the “Type” & “Choose from list” drop downs are both blank (no dopdowns)

I have 6 rooms and Bulbs, Sonos, Sockets configured.

node-red-contrib-dirigera v1.0.1
Node-RED version: v3.1.0
Node.js  version: v18.18.2

**Comment by zinen** [comment]
Based on your nice and detailed explanation I then assume you done everything correct and I dont see why that did not work. I have 5 steps I would like you to test out.
1. What error do you see if any in debug panel or the status text underneath the node?
2. Try to send and input into this via **NEW** inject node. What do you see?
3. Does your input contain a "non" recognized topic?
4. Could you while being the the editor restart your flow. And then watch the debug panel for any error messages. If the config did not work there should be an error message here when you first deployed. You can reproduce this by restarting. What error message do you see?
5. If nothing else comes up. Can you access the console and note errors on startup/restart?
![image](https://github.com/zinen/node-red-contrib-dirigera/assets/9595798/5ce0a5f0-183a-4901-800e-a06372596184)

**Comment by ukmoose** [comment]
What error do you see if any in debug panel or the status text underneath the node? 
 -  There are no errors in the debug panel, under the node or in the node-red log on first deploy
 
 Try to send and input into this via NEW inject node. What do you see?
- If I send a default new inject node message to the node I get the error message " getRoom cant find room name or id: -1" under the node, in the debug panel and console log

Does your input contain a "non" recognized topic?
-  If you mean the injected message any message results in the same error (even the two injects in the example)

Debug panel/ console on restart?
- Nothing odd. The usual info messages ending in the started flows message.

I also tried modifying line 23 in dirigera.js to change the debug level to 5 but no messages are returned when deleting the hub and repeating the process.

**Comment by zinen** [comment]
Its all very consistent with the config deployment going wrong somehow. Good thinking about edit the .js file, but could you try to edit line 67? at set debug=5. 

This should help revival where the error lies. I think the line 68 about testing the access code (login part) might not return the same result for you as it does in my setup. But lets see that you find out?

**Comment by zinen** [comment]
I just now pushed update `v1.0.2` to improve error catch when creating config. Maybe this will catch the error that happens to you.

**Comment by ukmoose** [comment]
Thanks!

I upgraded to 1.0.2 but still have the same issue

Here’s the console output:
22 Oct 15:04:54 - [info] Starting flows
running logIn
running fetch GET on URL: https://192.168.0.168:8443/v1/users/me
22 Oct 15:04:54 - [info] Started flows
22 Oct 15:04:54 - [info] [mqtt-broker:c0b3747d0339db2b] Connected to broker: mqtt://192.168.0.XXX:1883
Returned headers response:200
- api-version: 1.3.26
- connection: close
- content-length: 212
- content-type: application/json; charset=utf-8
- date: Sun, 22 Oct 2023 13:04:55 GMT
- etag: W/"d4-A9QwU/5JBtRK1/R/S/0tozvtzc4"
running getDevice
running fetch GET on URL: https://192.168.0.168:8443/v1/devices
Returned headers response:200
- api-version: 1.3.26
- connection: close
- content-length: 18133
- content-type: application/json; charset=utf-8
- date: Sun, 22 Oct 2023 13:05:13 GMT
- etag: W/"46d5-0HYWrj7yu3HBspxmrwG7v42Lm5Y"
running getDevice
running getDevice

If I add a new Inject node and attach it to the node on an inject I get the following

running getRoom
running getDevice
running fetch GET on URL: https://192.168.0.168:8443/v1/devices
Returned headers response:200
- api-version: 1.3.26
- connection: close
- content-length: 18133
- content-type: application/json; charset=utf-8
- date: Sun, 22 Oct 2023 13:07:52 GMT
- etag: W/"46d5-inPW/ONoGf4jY4r8OsqVX8qHwnw"
22 Oct 15:07:52 - [error] [dirigera:067e9b4e8c61cd77] getRoom cant find room name or id: -1


I’ve used my dirigera script on flows.nodered.org <http://flows.nodered.org/> to dump out https://192.168.0.168:8443/v1/devices
I don’t want to post it to GitHub (it contains the serial numbers of my Sonos gear and my homes lat / long ) but I can share it with you away from GitHub if it helps?

**Comment by zinen** [comment]
Ahh. You are that @ukmoose, I used your flow as basic for making this node. So thank you!

I see from your logs that the config indeed goes though and that your access token validates and answers back.  
```js
running fetch GET on URL: https://192.168.0.168:8443/v1/devices
Returned headers response:200
- api-version: 1.3.26
- connection: close
- content-length: 18133 // This mean theres plenty of data returned
- content-type: application/json; charset=utf-8 // This with the status code=200 means that all is okay and the retuned data is not som sort of error code or HTML returned.
```

So now I expect you will see the real problem in our browser. This node creates a API endpoint to transfer data from node-red server side to the client side dashboard. This means in my case that a URL of http://localhost:1880/ikeaDirigera/dirigera?nodeId=123343242blala(id of config node) is made accessible. When on the dashboard and open this node this API is queried. In case no deployed config is found then the returned message state that you should make and deploy first. So in your browser when you open this node - what error do you see in the console of the browser (e.g. developer tools its called in chrome)?

**Comment by ukmoose** [comment]
When the node has the hub configured, and I click to open the node I get the following error in ChromeConsole: 

<img width="880" alt="image" src="https://github.com/zinen/node-red-contrib-dirigera/assets/2537652/7b2de59d-7a9a-48c2-a150-314fcaa43a53">


vendor.js?v=:4 JQMIGRATE: Migrate is installed, version 3.3.0
red.min.js?v=:16 Node-RED: 3.1.0
red.min.js?v=:16 Versions
red.min.js?v=:16 Projects disabled
VM67:9 Uncaught TypeError: node.error is not a function
    at Object.success (<anonymous>:9:14)
    at c (vendor.js?v=:2:28294)
    at Object.fireWith [as resolveWith] (vendor.js?v=:2:29039)
    at l (vendor.js?v=:2:79800)
    at XMLHttpRequest.<anonymous> (vendor.js?v=:2:82254)
(anonymous) @ VM67:9
c @ vendor.js?v=:2
fireWith @ vendor.js?v=:2
l @ vendor.js?v=:2
(anonymous) @ vendor.js?v=:2
load (async)
send @ vendor.js?v=:2
ajax @ vendor.js?v=:2
s.ajax @ vendor.js?v=:4
S.<computed> @ vendor.js?v=:2
updateType @ VM67:5
oneditprepare @ VM67:78
t @ red.min.js?v=:18
R @ red.min.js?v=:18
open @ red.min.js?v=:18
n @ red.min.js?v=:19
show @ red.min.js?v=:19
edit @ red.min.js?v=:18
mt @ red.min.js?v=:18
yt @ red.min.js?v=:18
i @ vendor.js?v=:28
VM67:9 Uncaught TypeError: node.error is not a function
    at Object.success (<anonymous>:9:14)
    at c (vendor.js?v=:2:28294)
    at Object.fireWith [as resolveWith] (vendor.js?v=:2:29039)
    at l (vendor.js?v=:2:79800)
    at XMLHttpRequest.<anonymous> (vendor.js?v=:2:82254)
(anonymous) @ VM67:9
c @ vendor.js?v=:2
fireWith @ vendor.js?v=:2
l @ vendor.js?v=:2
(anonymous) @ vendor.js?v=:2
load (async)
send @ vendor.js?v=:2
ajax @ vendor.js?v=:2
s.ajax @ vendor.js?v=:4
S.<computed> @ vendor.js?v=:2
updateType @ VM67:5
(anonymous) @ VM67:83
dispatch @ vendor.js?v=:2
v.handle @ vendor.js?v=:2
trigger @ vendor.js?v=:2
(anonymous) @ vendor.js?v=:2
each @ vendor.js?v=:2
each @ vendor.js?v=:2
trigger @ vendor.js?v=:2
t @ red.min.js?v=:18
R @ red.min.js?v=:18
open @ red.min.js?v=:18
n @ red.min.js?v=:19
show @ red.min.js?v=:19
edit @ red.min.js?v=:18
mt @ red.min.js?v=:18
yt @ red.min.js?v=:18
i @ vendor.js?v=:28
VM67:9 Uncaught TypeError: node.error is not a function
    at Object.success (<anonymous>:9:14)
    at c (vendor.js?v=:2:28294)
    at Object.fireWith [as resolveWith] (vendor.js?v=:2:29039)
    at l (vendor.js?v=:2:79800)
    at XMLHttpRequest.<anonymous> (vendor.js?v=:2:82254)
(anonymous) @ VM67:9
c @ vendor.js?v=:2
fireWith @ vendor.js?v=:2
l @ vendor.js?v=:2
(anonymous) @ vendor.js?v=:2
load (async)
send @ vendor.js?v=:2
ajax @ vendor.js?v=:2
s.ajax @ vendor.js?v=:4
S.<computed> @ vendor.js?v=:2
updateType @ VM67:5
(anonymous) @ VM67:83
dispatch @ vendor.js?v=:2
v.handle @ vendor.js?v=:2
trigger @ vendor.js?v=:2
(anonymous) @ vendor.js?v=:2
each @ vendor.js?v=:2
each @ vendor.js?v=:2
trigger @ vendor.js?v=:2
(anonymous) @ red.min.js?v=:18
setTimeout (async)
_ @ red.min.js?v=:18
(anonymous) @ red.min.js?v=:18
t @ red.min.js?v=:18
R @ red.min.js?v=:18
open @ red.min.js?v=:18
n @ red.min.js?v=:19
show @ red.min.js?v=:19
edit @ red.min.js?v=:18
mt @ red.min.js?v=:18
yt @ red.min.js?v=:18
i @ vendor.js?v=:28

**Comment by zinen** [comment]
I see I fixed an error and then caused another. Updated to version `v1.0.3` just now. This basically should just throw the error to console and not actually fix your problem yet.  I'll test your device list later on my test client. Do you get a new error now?

I only have lights and blinds and so I figure that the problem could very well be in that your device list is different that mine that I tested with.

**Comment by ukmoose** [comment]
Updated to v.1.0.3
<img width="1549" alt="image" src="https://github.com/zinen/node-red-contrib-dirigera/assets/2537652/c1611128-7cf9-4003-8bc0-33f4e4578f3b">

In the debug window I now get "Dirigera get devices error: TypeError: Cannot read properties of undefined (reading 'id')" three times

**Comment by zinen** [comment]
Should be fixed in `v1.0.4` - The issue it seems was that not all your devices is assigned to a room. I added a no-room category to handle this.

Please do reply back with anything either working or still causing issues :)

**Comment by ukmoose** [comment]
Many thanks, it's working!
